### PR TITLE
Labeled framesets and related constants

### PR DESCRIPTION
**Pull request by SoupPotato**
Updated all Framesets with labels.

### CODE REVIEW

File: data/battle_anims/framesets.asm [hunk 1]
```diff
@@ -1,222 +1,222 @@
...
-	dw .Frameset_b0 ; BATTLEANIMFRAMESET_B0
-	dw .Frameset_b1 ; BATTLEANIMFRAMESET_B1
-	dw .Frameset_b2 ; BATTLEANIMFRAMESET_B2
-	dw .Frameset_b3 ; BATTLEANIMFRAMESET_B3
-	dw .Frameset_b4 ; BATTLEANIMFRAMESET_B4
-	dw .Frameset_b5 ; BATTLEANIMFRAMESET_B5
-	dw .Frameset_b6 ; BATTLEANIMFRAMESET_B6
-	dw .Frameset_b7 ; BATTLEANIMFRAMESET_B7
-	dw .Frameset_b8 ; BATTLEANIMFRAMESET_B8
+	dw .Frameset_hit_big               ; BATTLEANIMFRAMESET_HIT_BIG
```
Comment (vulcandth) [inline data/battle_anims/framesets.asm:4]: These labels, albeit local, point to an atomic chunk of code. So they should use .PascalCase and to also be consistent with the Frameset labels in `data/sprite_anims/framesets.asm`. We can leave the prefix `Framset_` with the underscore as this just aids in readability a little. This should be done for the rest as well.

```suggestion
    dw .Frameset_HitBig               ; BATTLEANIMFRAMESET_HIT_BIG
```

File: data/battle_anims/objects.asm [hunk 2]
```diff
@@ -20,379 +20,379 @@ BattleAnimObjects:
...
+	battleanimobj RELATIVE_X, $90, BATTLEANIMFRAMESET_BITE_1, BATTLEANIMFUNC_BITE, PAL_BATTLE_OB_GRAY, ANIM_GFX_CUT
 ; ANIM_OBJ_CUT_DOWN_LEFT
-	battleanimobj RELATIVE_X | X_FLIP, $ff, BATTLEANIMFRAMESET_3E, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_GRAY, ANIM_GFX_CUT
+	battleanimobj RELATIVE_X | X_FLIP, $ff, BATTLEANIMFRAMESET_CUT_DOWN_LEFT, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_GRAY, ANIM_GFX_CUT
 ; ANIM_OBJ_CUT_DOWN_RIGHT
-	battleanimobj RELATIVE_X | X_FLIP, $ff, BATTLEANIMFRAMESET_3F, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_GRAY, ANIM_GFX_CUT
-; ANIM_OBJ_CUT_UP_RIGHT
-	battleanimobj RELATIVE_X | X_FLIP, $ff, BATTLEANIMFRAMESET_40, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_GRAY, ANIM_GFX_CUT
+	battleanimobj RELATIVE_X | X_FLIP, $ff, BATTLEANIMFRAMESET_CUT_DOWN_RIGHT, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_GRAY, ANIM_GFX_CUT
+;  ANIM_OBJ_CUT_UP_RIGHT
```
Comment (vulcandth) [inline data/battle_anims/objects.asm:136]:
```suggestion
; ANIM_OBJ_CUT_UP_RIGHT
```

File: data/battle_anims/objects.asm [hunk 3]
```diff
@@ -20,379 +20,379 @@ BattleAnimObjects:
...
 ; ANIM_OBJ_RECOVER
-	battleanimobj RELATIVE_X, $90, BATTLEANIMFRAMESET_21, BATTLEANIMFUNC_RECOVER, PAL_BATTLE_OB_YELLOW, ANIM_GFX_BUBBLE
-; ANIM_OBJ_THUNDER1
-	battleanimobj RELATIVE_X | X_FLIP, $b0, BATTLEANIMFRAMESET_31, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_YELLOW, ANIM_GFX_LIGHTNING
-; ANIM_OBJ_THUNDER2
-	battleanimobj RELATIVE_X | X_FLIP, $b0, BATTLEANIMFRAMESET_32, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_YELLOW, ANIM_GFX_LIGHTNING
-; ANIM_OBJ_THUNDER3
-	battleanimobj RELATIVE_X | X_FLIP, $b0, BATTLEANIMFRAMESET_33, BATTLEANIMFUNC_NULL, PAL_BATTLE_OB_YELLOW, ANIM_GFX_LIGHTNING
+	battleanimobj RELATIVE_X, $90, BATTLEANIMFRAMESET_SMALL_BUBBLE, BATTLEANIMFUNC_RECOVER, PAL_BATTLE_OB_YELLOW, ANIM_GFX_BUBBLE
+; ANIM_OBJ_THUNDER_CENTER
```
Comment (vulcandth) [inline data/battle_anims/objects.asm:112]: You renamed some of the BattleAnimObjects in these comments... but you didn't actually update their constants with the new names.

**Review by vulcandth — CHANGES_REQUESTED** [review]
Great start; some feedback for now. I'll review it more when I get back home later.

File: constants/battle_anim_constants.asm [hunk 4]
```diff
@@ -306,191 +306,191 @@ DEF NUM_BATTLEANIMFUNCS EQU const_value
...
+	const BATTLEANIMFRAMESET_BEAM
+	const BATTLEANIMFRAMESET_BEAM_TIP
+	const BATTLEANIMFRAMESET_ICE_BUILDUP
+	const BATTLEANIMFRAMESET_FROZEN
+	const BATTLEANIMFRAMESET_CIRCLING_SPARKLE
+	const BATTLEANIMFRAMESET_THUNDER_CENTER
+	const BATTLEANIMFRAMESET_THUNDER_LEFT
+	const BATTLEANIMFRAMESET_THUNDER_RIGHT
+	const BATTLEANIMFRAMESET_THUNDER_WAVE_DISABLE
 	const BATTLEANIMFRAMESET_35
```
Comment (vulcandth) [inline constants/battle_anim_constants.asm:362]:
```suggestion
	const BATTLEANIMFRAMESET_THUNDER_WAVE_EXTRA
```

File: constants/battle_anim_constants.asm [hunk 5]
```diff
@@ -306,191 +306,191 @@ DEF NUM_BATTLEANIMFUNCS EQU const_value
...
+	const BATTLEANIMFRAMESET_POWDER_SNOW
+	const BATTLEANIMFRAMESET_RAZOR_LEAF_1
+	const BATTLEANIMFRAMESET_RAZOR_LEAF_2
+	const BATTLEANIMFRAMESET_EXPLOSION
+	const BATTLEANIMFRAMESET_BIG_ROCK
+	const BATTLEANIMFRAMESET_SMALL_ROCK
+	const BATTLEANIMFRAMESET_STRENGTH
+	const BATTLEANIMFRAMESET_SKULL_CROSSBONE
+	const BATTLEANIMFRAMESET_ACID
 	const BATTLEANIMFRAMESET_1E
```
Comment (Rangi42) [inline constants/battle_anim_constants.asm:339]: If this were used, does it have a sensible name? Has it been useful in any custom anims?
```suggestion
	const BATTLEANIMFRAMESET_1E ; unused
```

File: constants/battle_anim_constants.asm [hunk 6]
```diff
@@ -306,191 +306,191 @@ DEF NUM_BATTLEANIMFUNCS EQU const_value
...
+	const BATTLEANIMFRAMESET_LOCK_ON_1
+	const BATTLEANIMFRAMESET_LOCK_ON_2
+	const BATTLEANIMFRAMESET_LOCK_ON_3
+	const BATTLEANIMFRAMESET_LOCK_ON_4
+	const BATTLEANIMFRAMESET_MIND_READER_1
+	const BATTLEANIMFRAMESET_MIND_READER_2
+	const BATTLEANIMFRAMESET_MIND_READER_3
+	const BATTLEANIMFRAMESET_MIND_READER_4
+	const BATTLEANIMFRAMESET_SAFEGUARD
+	const BATTLEANIMFRAMESET_UNUSED_MEGAPHONE
```
Comment (Rangi42) [inline constants/battle_anim_constants.asm:461]: Following the pattern of other `git grep -E 'const .+UNUSED'` names.
```suggestion
	const BATTLEANIMFRAMESET_MEGAPHONE_UNUSED
```

File: constants/battle_anim_constants.asm [hunk 7]
```diff
@@ -306,191 +306,191 @@ DEF NUM_BATTLEANIMFUNCS EQU const_value
...
+	const BATTLEANIMFRAMESET_CURSE_NAIL
+	const BATTLEANIMFRAMESET_FORESIGHT_SHINE
+	const BATTLEANIMFRAMESET_RAPID_SPIN
+	const BATTLEANIMFRAMESET_SWAGGER
+	const BATTLEANIMFRAMESET_MEAN_LOOK
+	const BATTLEANIMFRAMESET_UPSIDE_DOWN_PUNCH
+	const BATTLEANIMFRAMESET_GROWING_SPARKLE
+	const BATTLEANIMFRAMESET_RAIN
+	const BATTLEANIMFRAMESET_PSYCH_UP
 	const BATTLEANIMFRAMESET_B2
```
Comment (Rangi42) [inline constants/battle_anim_constants.asm:487]: Same for this, does it have a descriptive name?

File: data/battle_anims/framesets.asm [hunk 8]
```diff
@@ -1,222 +1,222 @@
...
+	dw .Frameset_ParalyzedFlipped   ; BATTLEANIMFRAMESET_PARALYZED_FLIPPED
+	dw .Frameset_Haze               ; BATTLEANIMFRAMESET_HAZE
+	dw .Frameset_Mist               ; BATTLEANIMFRAMESET_MIST
+	dw .Frameset_Horn               ; BATTLEANIMFRAMESET_HORN
+	dw .Frameset_Needle             ; BATTLEANIMFRAMESET_NEEDLE
+	dw .Frameset_Flower             ; BATTLEANIMFRAMESET_FLOWER
+	dw .Frameset_BarrageBall        ; BATTLEANIMFRAMESET_BARRAGE_BALL
+	dw .Frameset_PayDay             ; BATTLEANIMFRAMESET_PAY_DAY
+	dw .Frameset_Heart              ; BATTLEANIMFRAMESET_HEART
+	dw .Frameset_SprinningBone      ; BATTLEANIMFRAMESET_SPINNING_BONE
```
Comment (Rangi42) [inline data/battle_anims/framesets.asm:123]:
```suggestion
	dw .Frameset_SpinningBone       ; BATTLEANIMFRAMESET_SPINNING_BONE
```

**Comment by mid-kid** [comment]
Just gonna comment that I like the naming scheme :)
So if I don't get around to properly reviewing this in the next week, consider this an approval.

**Review by Rangi42 — APPROVED** [review]

**Comment by vulcandth** [comment]
Once I work out the conflicts when merging this to pokegold; i'll merge this.

**Pull request merged**